File: Net/01/TasksManager.Application/ToDoTasksEventHandler.cs
using Commons.Tasks;
using Cqrs;
using log4net;
using TasksManager.Repositories;

namespace TasksManager
{
    public class ToDoTasksEventHandler : IMessageHandler
    {
        private static ILog _logger = LogManager.GetLogger(typeof(ToDoTasksEventHandler));
        private IRepository<ToDoTaskDao> _repository;
        private IBus _bus;

        public void Register(IBus bus)
        {
            _bus = bus;
            _bus.RegisterHandler(m => Handle((TaskCreated)m), typeof(TaskCreated));
            _bus.RegisterHandler(m => Handle((TaskPriorityChanged)m), typeof(TaskPriorityChanged));
            _bus.RegisterHandler(m => Handle((TaskTitleChanged)m), typeof(TaskTitleChanged));
            _bus.RegisterHandler(m => Handle((TaskCompleted)m), typeof(TaskCompleted));
        }

        public ToDoTasksEventHandler(IRepository<ToDoTaskDao> repository)
        {
            _repository = repository;
        }

        public void Handle(TaskCreated message)
        {
            _logger.Info("TaskCreated");
            var toDoTask = new ToDoTaskDao
            {
                Id = message.Id,
                CreationDate = message.CreationDate
            };
            _repository.Save(toDoTask);
            Handle(message.TitleSet);
            Handle(message.PrioritySet);
        }

        public void Handle(TaskPriorityChanged message)
        {
            _logger.Info("TaskPriorityChanged");
            var toDoTask = _repository.GetById(message.Id);
            toDoTask.Priority = message.New;
            _repository.Save(toDoTask);
        }

        public void Handle(TaskTitleChanged message)
        {
            _logger.Info("TaskTitleChanged");
            var toDoTask = _repository.GetById(message.Id);
            toDoTask.Title = message.New;
            _repository.Save(toDoTask);
        }

        public void Handle(TaskCompleted message)
        {
            _logger.Info("TaskCompleted");
            _repository.Delete(message.Id);
        }
    }
}

File: Java/02/tasks-manager-application/src/main/java/org/tasksmanager/ToDoTasksEventHandler.java
package org.tasksmanager;

import java.util.UUID;
import java.util.logging.Level;
import java.util.logging.Logger;
import javax.inject.Inject;
import javax.inject.Named;
import org.commons.Tasks.*;
import org.cqrs.*;
import org.tasksmanager.Repositories.*;

@Named("toDoTasksEventHandler")
public class ToDoTasksEventHandler implements MessageHandler {

    private static final Logger logger = Logger.getLogger(ToDoTasksEventHandler.class.getSimpleName());
    private final Repository<ToDoTaskDao,UUID> _repository;
    private Bus _bus;

    @Override
    public void Register(Bus bus) {
        _bus = bus;
        _bus.RegisterHandler(m -> Handle((TaskCreated) m), TaskCreated.class);
        _bus.RegisterHandler(m -> Handle((TaskPriorityChanged) m), TaskPriorityChanged.class);
        _bus.RegisterHandler(m -> Handle((TaskTitleChanged) m), TaskTitleChanged.class);
        _bus.RegisterHandler(m -> Handle((TaskCompleted) m), TaskCompleted.class);
    }

    @Inject
    public ToDoTasksEventHandler(Repository<ToDoTaskDao,UUID> toDoTasksRepository) {
        _repository = toDoTasksRepository;
    }

    public void Handle(TaskCreated message) {
        logger.log(Level.INFO, "TaskCreated");
        ToDoTaskDao toDoTask = new ToDoTaskDao();
        toDoTask.setId(message.getId());
        toDoTask.setCreationDate(message.getCreationDate());

        _repository.Save(toDoTask);
        Handle(message.getTitleSet());
        Handle(message.getPrioritySet());
    }

    public void Handle(TaskPriorityChanged message) {
        logger.log(Level.INFO, "TaskPriorityChanged");
        ToDoTaskDao toDoTask = _repository.GetById(message.getId());
        toDoTask.setPriority(message.getNew());
        _repository.Save(toDoTask);
    }

    public void Handle(TaskTitleChanged message) {
        logger.log(Level.INFO, "TaskTitleChanged");
        ToDoTaskDao toDoTask = _repository.GetById(message.getId());
        toDoTask.setTitle(message.getNew());
        _repository.Save(toDoTask);
    }

    public void Handle(TaskCompleted message) {
        logger.log(Level.INFO, "TaskCompleted");
        _repository.Delete(message.getId());
    }
}
